src/Rx/Weno/PatientPharmacyController.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Sherwin Gaddis <sherwingaddis@gmail.com>
+ * @copyright Copyright (c )2019. Sherwin Gaddis <sherwingaddis@gmail.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Rx\Weno;
+
+use OpenEMR\Common\Database\Connector;
+use OpenEMR\Entities\PatientData;
+
+class PatientPharmacyController
+{
+    private $entityManager;
+
+    private $repository;
+
+    public function __construct()
+    {
+        $database = Connector::Instance();
+        $entityManager = $database->entityManager;
+        $this->repository = $entityManager->getRepository(PatientData::class);
+    }
+
+    public function updatePatientPharmacyApi($id, $pid)
+    {
+        $patientPharmacy = $this->repository;
+        $set = $patientPharmacy->updatePatientPharmacy($id, $pid);
+        return $set;
+    }
+}
+
+if (!empty($_GET['id'])) {
+    $id = filter_input(INPUT_GET, 'id', FILTER_VALIDATE_INT);
+    $pid = $GLOBALS['pid'];
+    $set = new PatientPharmacyController();
+    echo $set->updatePatientPharmacyApi($id, $pid);
+} else {
+    echo json_encode(["error" => "No id sent!"]);
+}
```

Comment:
This block of code can't exist with the class script. Let me know if you need guidance how to migrate this out of this class script.

Reply:
Yes, I need your assistance to show me how this would be changed. It is not in the class. It is outside of the class. It is calling the class after an ajax call to this controller.

Reply:
Recommend separating this into a script like prescriptionorder.php.